### ISSUE TITLE
Review url being optional for SplashRequest

### ISSUE THREAD

**Issue by Gallaecio** [scrapy-plugins/scrapy-splash#270]
As part of https://github.com/scrapy-plugins/scrapy-splash/pull/269, the `url` parameter to `SplashRequest` is no longer optional.

@elacuesta noticed that this is a backward-incompatible change. Moreover, the upcoming Scrapy 2.4 will make this change unnecessary (https://github.com/scrapy/scrapy/pull/4835).

We should consider reverting that part of https://github.com/scrapy-plugins/scrapy-splash/pull/269.

**Comment by lopuhin** [comment]
sounds good to me, thanks for the heads up @Gallaecio

**Comment by Rafiot** [comment]
Reverting https://github.com/scrapy-plugins/scrapy-splash/pull/269/commits/4154782b16c6248a5a9b8fe3ed0ebb1cbc600a60 will do just that, so just go for it.

**Comment by protokoul** [comment]
Hi. Is this issue still relevant to the latest version of scrapy-splash? I reverted the changes in https://github.com/scrapy-plugins/scrapy-splash/commit/4154782b16c6248a5a9b8fe3ed0ebb1cbc600a60, ran the tests locally and no failures were noticed.

**Comment by Gallaecio** [comment]
Well, I think reverting that commit is precisely the goal. I assume you reverted the entire commit, nor just the non-test code, and that keeping the current test code and applying the rest of the commit would result in broken tests.

**Comment by protokoul** [comment]
Yes, I reverted the entire commit including the changes to the tests.

**Comment by Gallaecio** [comment]
My understanding is that, reverting the commit, [scrapy-splash would fail with scrapy>=2,<2.4](https://github.com/scrapy-plugins/scrapy-splash/pull/269#issuecomment-703901740) if the URL is omitted, because the default URL uses the unsupported `about:` protocol.

The easiest solution I think would be to revert that commit and require scrapy>=2.4.
